api/src/main/kotlin/io/spine/protodata/renderer/SourceFileSet.kt

```diff
@@ -364,18 +366,18 @@ public class FileLookup(
 /**
  * Part of the fluent API for creating new source files.
  */
-public class FileCreation(
+public class FileCreation<N: ProtoDeclarationName>(
     private val sources: SourceFileSet,
-    private val name: TypeName
+    private val name: N
 ) : FileOperation {
 
     /**
      * Attempts to create a file path for the given type name using the given [convention].
      *
      * If the convention does not define a declaration for the given type, returns `null`.
      */
-    public fun <L : Language, N : TypeNameElement<L>> namedUsing(
-        convention: TypeConvention<L, N>
+    public fun <L : Language, T : NameElement<L>> namedUsing(
```

Comment:
This signature is now more complex than before.

Let's document the generic parameter meaning here as well.